python/mxnet/model.py

```diff
@@ -286,48 +450,22 @@ def _train_multi_device(symbol, ctx, arg_names, param_names, aux_names,
         while True:
             do_reset = True
             for data_batch in train_data:
-                _load_data(data_batch, data_arrays)
-                _load_label(data_batch, label_arrays)
+
+                executor_manager.load_data_batch(data_batch)
 
                 if monitor is not None:
```

Comment:
I did think about this, but my conclusion was to unwrap the optional value as quickly as possible. Also thinking about it as an external API I didn't make sense to have one that does nothing on None inputs. That being said, I can change it if you would wish.